src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -5073,16 +5070,26 @@ bool StorageReplicatedMergeTree::getFakePartCoveringAllPartsInPartition(const St
 void StorageReplicatedMergeTree::restoreMetadataInZooKeeper()
 {
     LOG_INFO(log, "Restoring replica metadata");
+
+    using enum InitializationPhase;
+    {
+        std::lock_guard lock(initialization_mutex);
+        if (init_phase == INITIALIZING || init_phase == STARTUP_IN_PROGRESS)
+            throw Exception(ErrorCodes::NOT_INITIALIZED, "Table is not initialized yet");
```

Comment:
It's a bad check. 
Restore tries to start the table after it finishes, so this check tried to avoid AttachThread running `startup` method at the same time. `INITIALIZATION_DONE` and `STARTUP_DONE` are both states when we know that attach thread is finished.
I think I can remove `STARTUP_*` states.